### ISSUE TITLE
Chapter Titles in the Alt Status Bar

### ISSUE THREAD

**Issue by offset-torque** [koreader/koreader#9042]
**Context:**
- Chapter titles are useful for acknowledging the reading context especially when reading several books and switching between them
- Bottom status bar has **chapter titles** but it is generally crowded with several items which can only be added to the bottom bar. Not much space is available for chapter titles.

**Proposal:**
If it is not too much work, I think having the option to have **Chapter Titles** in the **Alt Status Bar** will be useful

**Benefits:**
- This option can free up precious bottom bar space
- Some chapter titles can be very long so they may need wider space to fit, **Alt status bar** is ideal for this
- It will look more natural because printed books generally have **chapter title** and **page number** in the header
- BONUS: If bottom bar is hidden while using this feature, screen will have a clean "printed book" feeling

![chapter title](https://user-images.githubusercontent.com/84420582/165911378-957300cb-505f-4f2a-bc5d-c37cb5d73283.jpeg)

**Comment by zwim** [comment]
Sorry, the top bar is handled by crengine. Changes there are very difficult and subtile.

Also, I dont think we have the infos right away in crengine, what the title is (as we also provide an alternate toc in KOReader).

So if you would ask me, I dont see a clear way to do it. It would really be a lot of work (and I personally would not use it, and have no time for that).

If you say the bottom bar is crowded, why not disable some items? Or maybe someone (no me, not interested) implements a double line status bar, tripple line ... until there is more info than text left ;)

But as always, PRs are welcome.

**Comment by offset-torque** [comment]
I see, thanks for the answer zwim. I thought they are showing the same info, but in different positions, so my proposal shouldn't be much trouble. Too bad Alt Status Bar is legacy code. That area could be really useful.

> If you say the bottom bar is crowded, why not disable some items?

Yeah I will do that now since this is the only option.

> double line status bar, tripple line ... until there is more info than text left ;)

Actually that's not the solution. Many e-book readers on Android handle that extra info elegantly with a single tap without cluttering reading interface (probably ios apps too). If you have ever used Moon Reader or something similar, probably you know what I mean.

I see there is a `wontfix` tag. 
Should I leave the issue open ? 
What is the procedure in this case ?

**Comment by zwim** [comment]
You have opened it, you can close it at any time (if it is solved for you).

You can leave the issue open for some time, maybe someone sees it and wants to do a PR.

**Comment by mergen3107** [comment]
> Many e-book readers on Android handle that extra info elegantly with a single tap without cluttering reading interface

Have you tried using “pages” of the bottom status bar? You click once on the status bar to go through pages and the last one hides it. It is somewhere in the settings of the status bar, forgot the exact name

**Comment by zwim** [comment]
Could/would a gesture to show some book information (let say title, author, chapter title, pages / pages read ...) help you? 

If you are not sure in which book you are actually -> gesture -> InfoMessage -> dismiss voila.

**Comment by offset-torque** [comment]
Thanks for reminding me of this, **mergen3107**. I tried it but even with a few items, it is a hassle to cycle through them. I remembered why I forgot about this feature :)

---
Thanks for asking **zwim**. I am currently using `Show ToC` gesture to see where I am. It is already very easy for me (just one multiswipe) so I think any dev effort would be wasted on this. I opened this issue because I thought it can be achieved with a "very low effort" change to the Alt Status Bar and the result would be aesthetically pleasing.

After reading your last suggestion, now I noticed that, ToC doesn't contain the `Book Title`. If it is added there, ToC can also serve the purpose of **"Which book I am in ?"** in addition to **"Which chapter I am in ?"**. 

Because some books have very non-descriptive chapter names (mostly academic books), especially if they are converted from another format. So it might be a good idea to have the **Book Title** in the ToC.
